scripts/generate-docs.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * This script generates the SDK reference documentation based on the methods array defined at the beginning of the file.
+ * For a generated example, please refer to https://avaprotocol.org/docs/ethereum/develop/js-sdk
+ *
+ * Usage:
+ * node ./scripts/generate-docs.js
+ */
+
+const fs = require("fs");
+
+const OUTPUT_DIR = "./scripts"; // The directory to output the generated documentation; setting is to ./scripts for now.
+const OUTPUT_FILENAME = "SDK_References.md";
+
+const methods = [
+  {
+    name: "authWithAPIKey",
```

Comment:
I would not publicly document `authWithAPIKey` for now. Considering it's a private api for now. There is no way for an end-user to acquire this key. Once we implement an oauth process where application will register to get app id, and api key then we can start annoucen this. right now, no one can actually use this or even shoulduse this.

Reply:
Right, that’s a good call. I removed it from this PR.